### DISCUSSION TITLE
e2e test on pre-push hook

### DISCUSSION THREAD

**Discussion by daiagi** [kodadot/nft-gallery#10249] [General]
@prury   @kodadot/internal-dev 

I think it could be useful to have playwright tests run on pre-push hook

avoid wasting time of waiting for gh to run them

**Comment by prury** [comment]
> @prury @kodadot/internal-dev
> 
> I think it could be useful to have playwright tests run on pre-push hook
> 
> avoid wasting time of waiting for gh to run them

if i understood correctly, running them on pre push hook would be running the test before commits goes to GH.
i'm not sure if it will be faster since it will need to run the webserver locally, test it, and them allow push to go trough or not
wdyt?

**Comment by Jarsen136** [comment]
> I think it could be useful to have playwright tests run on pre-push hook
> 

It will take several minutes (or even more) to run the test whenever pushing something. There is also some waste of time.

![image](https://github.com/kodadot/nft-gallery/assets/31397967/5a396717-79ae-4d2f-92bb-d87829ff1ec0)

**Comment by preschian** [comment]
Running the test on our devices will help reduce the duration. Since our machines are more powerful than the CI

Ref: https://github.com/kodadot/nft-gallery/issues/3123#issuecomment-1426636876. Plus, if we use cache in our monorepo. We can use one of these:
- turborepo cache, or
- [nx cloud](https://nx.app/)

**Reply by prury** [reply]
below its our slowest test: 
![image](https://github.com/kodadot/nft-gallery/assets/36627808/b47fb9df-d245-4644-a817-d223220cf914)

3m42s(building app + installing browsers + testing)

today they all run on a matrix that split tests between runners and logs are easy to see(pretty similar to nx cloud)


> Running the test on our devices will help reduce the duration. Since our machines are more powerful than the CI
> 

not so sure, as running playwright locally with more workers enabled always throw some extra fails

you can all try it by running npx playwright test --workers x 
x = amount
